[wip][jit] Refactor math ops to use custom op bindings

driazati opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#19373 [jit] Refactor math ops to use custom op bindings**

Differential Revision: [D14987159](https://our.internmc.facebook.com/intern/diff/D14987159)

~~Now that we have a way to specify basic aliasing behavior for custom ops, we can register math ops as custom ops (and hopefully more things in register_prim_ops). Due to this we don't need to manually whitelist `aten` and `prim` for alias anaylsis, so the restriction when binding custom ops can be deleted.~~

TODO: right now aten ops are assumed to be pure and custom ops are assumed to be side effectful. we should merge these so everything is assumed to be side effectful unless stated otherwise

Commit:
[jit] Refactor math ops to use custom op bindings

Commit:
Update on "[jit] Refactor math ops to use custom op bindings"

[jit] Refactor math ops to use custom op bindings

gh-metadata: pytorch pytorch 19373 gh/driazati/13/head

Commit:
Update on "[jit] Refactor math ops to use custom op bindings"

[jit] Refactor math ops to use custom op bindings

gh-metadata: pytorch pytorch 19373 gh/driazati/13/head

eellison commented on torch/csrc/jit/custom_operator.h:
```diff
@@ -182,26 +182,6 @@ Operator createOperator(
 
   auto schema = torch::jit::detail::inferAndCheckSchema<Traits>(schemaOrName);
 
-  // [custom operator aliasing] Currently, we have no way for the user to
```
Why did you delete this ?

driazati replied:
#18589 makes it unnecessary

suo replied:
I think we should do this clean up in a separate PR where you track down all references to this note and fix those up as well. Don't leave the pointers dangling!

Commit:
Update on "[jit] Refactor math ops to use custom op bindings"

[jit] Refactor math ops to use custom op bindings

gh-metadata: pytorch pytorch 19373 gh/driazati/13/head
